transcom/mymove
# B 18544 sc create safety move

## 18544
## Service Counselor can create a Safety Move

Agility link 
https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A897666
INT PR
https://github.com/transcom/mymove/pull/12744


This story also has a FF and should go with code checkin for milmove-feature-flag branch

### Setup to Run the Code

You will need to work with the FF FEATURE_FLAG_SAFETY_MOVE.
You will need to have a Service Counselor with safety privileges and one without privileges.

### How to test

1.  Login as a Service Counselor that has Safety Move privileges, make sure the FEATURE_FLAG_SAFETY_MOVE is on.
2. Access the Service Counselor's Customer Search tab, search for a user name that doesn't exist, and click Add Customer.
3. Verify that you see a new question as if this is a Safety Move.
4. Click the Yes radio button and verify that the DoDID is disabled and Okta and CAC questions at the bottom of the page are no longer visible.
5. Fill out the required information and click Save.
6. This should route you to the create order/move page.  Verify that Safety Move is selected and you can save successfully.
7. Run through 2-6 again, but pick No to the Safety Move question and verify that you can type in DoDID and that you see and have to fill out the Okta and CAC questions.
8. Verify that you do NOT see a choice of Safety Moves in the Orders type dropdown on the next page (since you said No to the question on the previous page).
9. Navigate back to the Customer search tab, type in a customer that already exists, click Create new move button.  Verify that you can create a Safety move and save here.
10. Login with a SC that does not have Safety Move privileges, run through the create new customer again and verify that you do not see the Safety Move question or the Safety Move choice in the Orders type dropdown.
11. Turn the FF off and login with the first user in number 1 above and verify that you do not see the Safety Move question on create new user or in the Orders type dropdown.
